doc/docs/Python_Tutorials/Near_to_Far_Field_Spectra.md

```diff
@@ -635,6 +635,350 @@ if __name__ == "__main__":
+    fig, ax = plt.subplots(subplot_kw={"projection": "polar"}, figsize=(6, 6))
+    ax.plot(
+        theta_rad,
+        radial_flux,
+        "b-",
+    )
+    ax.set_theta_direction(-1)
+    ax.set_theta_offset(0.5 * math.pi)
+    ax.set_thetalim(0, 0.5 * math.pi)
+    ax.grid(True)
+    ax.set_rlabel_position(22)
+    ax.set_ylabel("radial flux (a.u.)")
+    ax.set_title("radiation pattern in polar coordinates")
+
+    if mp.am_master():
+        fig.savefig(
+            "disc_radiation_pattern_polar.png",
+            dpi=150,
+            bbox_inches="tight",
+        )
+
+
+def plot_radiation_pattern_3d(theta_rad: np.ndarray, radial_flux: np.ndarray):
+    """Plots the radiation pattern in 3d Cartesian coordinates.
+
+    Args:
+      theta_rad: angles of the radiation pattern.
+      radial_flux: radial flux of the far fields in polar coordinates.
+    """
+    phis = np.linspace(0, 2 * np.pi, num_angles)
+
+    xs = np.zeros((len(theta_rad), len(phis)))
+    ys = np.zeros((len(theta_rad), len(phis)))
+    zs = np.zeros((len(theta_rad), len(phis)))
+
+    for i, theta in enumerate(theta_rad):
+        for j, phi in enumerate(phis):
+            xs[i, j] = radial_flux[i] * np.sin(theta) * np.cos(phi)
+            ys[i, j] = radial_flux[i] * np.sin(theta) * np.sin(phi)
+            zs[i, j] = radial_flux[i] * np.cos(theta)
+
+    fig, ax = plt.subplots(subplot_kw={"projection": "3d"}, figsize=(6, 6))
+    ax.plot_surface(xs, ys, zs, cmap="inferno")
+    ax.set_title("radiation pattern in 3d")
+    ax.set_box_aspect((np.amax(xs), np.amax(ys), np.amax(zs)))
+    ax.set_zlabel("radial flux (a.u.)")
+    ax.set(xticklabels=[], yticklabels=[])
+
+    if mp.am_master():
+        fig.savefig(
+            "disc_radiation_pattern_3d.png",
+            dpi=150,
+            bbox_inches="tight",
+        )
+
+
+def radiation_pattern(
+    theta_rad: np.ndarray, sim: mp.Simulation, n2f_mon: mp.DftNear2Far
+) -> np.ndarray:
+    """Computes the radiation pattern from the near fields.
+
+    Args:
+      theta_rad: angles of the radiation pattern.
+      sim: a `Simulation` object.
+      n2f_mon: a `DftNear2Far` object returned by `Simulation.add_near2far`.
+    """
+    e_field = np.zeros((theta_rad.shape[0], 3), dtype=np.complex128)
+    h_field = np.zeros((theta_rad.shape[0], 3), dtype=np.complex128)
+    for n in range(num_angles):
+        far_field = sim.get_farfield(
+            n2f_mon,
+            mp.Vector3(
+                FARFIELD_RADIUS_UM * math.sin(theta_rad[n]),
+                0,
+                FARFIELD_RADIUS_UM * math.cos(theta_rad[n]),
+            ),
+        )
+        e_field[n, :] = [np.conj(far_field[j]) for j in range(3)]
+        h_field[n, :] = [far_field[j + 3] for j in range(3)]
+
+    flux_r = np.real(e_field[:, 1] * h_field[:, 2] - e_field[:, 2] * h_field[:, 1])
+    flux_z = np.real(e_field[:, 0] * h_field[:, 1] - e_field[:, 1] * h_field[:, 0])
+    flux_rz = np.sqrt(np.square(flux_r) + np.square(flux_z))
```

Comment:
```suggestion
    flux_rz = flux_r * np.sin(theta_rad) + flux_z * np.cos(theta_rad)
```
you should be taking the dot product with `n̂` here.  (Though I think it should give the same results, since at a large radius the flux should be almost all radially outwards, I think?)

Reply:
I verified that this suggested change does not in fact change the results.